### ISSUE TITLE
[Question] Is accessing a VPC resource from AWS Account A possible for a RDSStack that is in AWS Account B?

### ISSUE THREAD

**Issue by brifiction** [serverless-stack/sst#1934]
Hello @fwang, is it possible to attach a VPC resource from AWS Account A, onto a RDS Construct that will be deployed under AWS Account B?

Please see below example `RDSStack` for better context:

```javascript
  // Same Organisational Unit, under accountA@company.com - built and deployed from another Stack
  const testVPCLookup = ec2.Vpc.fromLookup(
    stack,
    "Test-VPC-External",
    {
      vpcId: "vpc-1234567890abcd",
      isDefault: false,
    }
  );

// Same Organisational Unit, under accountB@company.com - built and deployed from this Stack
const testDBInstance = new rds.DatabaseInstance(
      stack,
      `TestDbInstance`,
      {
        vpc: testVPCLookup,
        vpcSubnets: {
          subnetType: ec2.SubnetType.PUBLIC,
        },
        engine: rds.DatabaseInstanceEngine.mysql({
          version: rds.MysqlEngineVersion.VER_8_0_23,
        }),
        instanceType: ec2.InstanceType.of(
          ec2.InstanceClass.BURSTABLE3,
          ec2.InstanceSize.MICRO
        ),
        credentials: rds.Credentials.fromGeneratedSecret('is-fetching-vpc-from-another-account-possible'),
        multiAz: false,
        allocatedStorage: 20,
        maxAllocatedStorage: 25,
        allowMajorVersionUpgrade: false,
        autoMinorVersionUpgrade: true,
        backupRetention: cdk.Duration.days(7),
        deleteAutomatedBackups: true,
        removalPolicy: cdk.RemovalPolicy.RETAIN,
        deletionProtection: true,
        databaseName: `mysql8-${stack.stage}`,
        publiclyAccessible: true,
      }
```

Lastly, will 'cross-account resource fetching' be a new feature in `serverless-stack`? If so, this will be a great feature to have, like using RAM (Resource Access Manager) with `aws-cdk` features to access Shared Resources.

**Comment by brifiction** [comment]
Apologies @fwang , there was a bug during the usage of `vpcId` property at the time of this post. And note, I've already set up a Shared Resource in the RAM, to share VPCs for use here in `serverless-stack` snippet above.

The steps taken were:

1. First, proceeding to AWS Management Console, heading over to RAM and 'Create resource share' for VPCs.
2. Next, once created, perform a simple test with `aws-cli` with command(s) needed to fetch the 'shared' VPC - this test is to ensure that the VPC found was successful.
3. Lastly once confirmed test successful, jumping back into codebase and retry the `vpcId` property again. In this case, must ensure that the targeted VPC is default = false.

In summary, unsure what happened with the `vpcId` lookup approach, despite RAM resource share pre-requisite steps completed - but now the `sst build` is successful.
